### DISCUSSION TITLE
请问一下如何通过MessageSegment给QQ发送带透明通道的图片？

### DISCUSSION THREAD

**Discussion by okashi-ya** [nonebot/discussions#387] [问答]
我的代码大概是这个样子的：

io = BytesIO()
pic_image.save(io, format="png")
pic_image.save("000.png")  #将图片保存到文件中 是带透明通道的
dst_msg = MessageSegment.image(io) 
message = ("图片：" + dst_msg)
后面是发送逻辑

图片发送后，发现图片透明通道丢失，且右键保存时图片类型为jpg。但我直接将image保存下来为png的时候是正常的。
请问我如何通过MessageSegment.image发送一个带透明通道的图片？

**Comment by synodriver** [comment]
qq压缩的吧？

**Reply by okashi-ya** [reply]
感觉不是，我这个图直接发送出去是正常的，而且右键点击另存为时，显示格式是png。
但是我通过MessageSegment.image发送的图片，透明的地方全是黑色的，然后右键另存为时，显示格式为jpg。

**Reply by yanyongyu** [reply]
这个问题请去协议端提问，与nonebot无关，nonebot不会对原数据进行任何处理，直接发送

**Comment by synodriver** [comment]
emmm……这个似乎不应该怪nonebot，这种方式实际上给onebot那边是一个base64，得看具体协议端怎么实现的
